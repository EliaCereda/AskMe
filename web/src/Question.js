import React, {Component} from "react";

class Question extends Component {
    render() {
        return (
            <article className="Question">
                <header>
                    <h3>{this.props.title}</h3>
                    <h4>opened by {this.props.user}</h4>
                </header>

                <p>{this.props.body}</p>
            </article>
        );
    }
}

export default Question;
