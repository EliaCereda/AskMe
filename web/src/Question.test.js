import React from 'react';
import Question from "./Question";

describe('Question', () => {
    it('renders correctly', () => {
        const question = {
            title: 'titolo della domanda',
            body: 'corpo della domanda',
            user: 'utente'
        };

        const wrapper = shallow(<Question {...question} />);
        expect(wrapper).toMatchSnapshot();
    });
});
