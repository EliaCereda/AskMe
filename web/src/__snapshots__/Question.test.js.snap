// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders correctly 1`] = `
<article
  className="Question"
>
  <header>
    <h3>
      titolo della domanda
    </h3>
    <h4>
      opened by 
      utente
    </h4>
  </header>
  <p>
    corpo della domanda
  </p>
</article>
`;
